### ISSUE TITLE
server crashes when user attempts to log in.

### ISSUE THREAD

**Issue by sudoman** [processone/ejabberd#1486]
> What version of ejabberd are you using?

16.01-2

> What operating system (version) are you using?

Trisquel 8, which is based off of Ubuntu 16.04.

> How did you install ejabberd (source, package, distribution)?

I installed the Trisquel 8 packaged version. I imported the user data from an older server, approximately version 2.1.11.

> What did not work as expected? Are there error messages in the log? What
> was the unexpected behavior? What was the expected result?

A user was unable to log in because the server crashes when they try to do so.

````
 2017-01-23 22:58:43 =ERROR REPORT====
** State machine <0.20956.69> terminating
** Last message in was
{send_filtered,{pep_message,<<"http://jabber.org/protocol/nick+notify">>},{jid,<<"user_person">>,<<"example.com">>,<<>>,<<"user_person">>,<<"example.com">>,<<>>},{jid,<<"user_person">>,<<"example.com">>,<<"mox">>,<<"user_person">>,<<"example.com">>,<<"mox">>},{xmlel,<<"message">>,[{<<"type">>,<<"headline">>}],[{xmlel,<<"event">>,[{<<"xmlns">>,<<"http://jabber.org/protocol/pubsub#event">>}],[{xmlel,<<"items">>,[{<<"node">>,<<"http://jabber.org/protocol/nick">>}],[{xmlel,<<"item">>,[{<<"id">>,"5704CDFE9EEEC"}],[{xmlelement,"nick",[{"xmlns","http://jabber.org/protocol/nick"}],[{xmlcdata,<<"Tom
Fiddle">>}]}]}]}]},{xmlel,<<"delay">>,[{<<"xmlns">>,<<"urn:xmpp:delay">>},{<<"from">>,<<"user_person@example.com/9951760901392301947684259">>},{<<"stamp">>,<<"2014-02-13T18:07:58Z">>}],[{xmlcdata,<<>>}]}]}}
** When State == session_established
** Data ==
{state,{socket_state,p1_tls,{tlssock,#Port<0.1809927>,#Port<0.1810780>},<0.21529.69>},ejabberd_socket,#Ref<0.0.114819073.134899>,false,<<"7963476858449577076">>,undefined,c2s,c2s_shaper,false,true,true,true,[verify_none,compression_none,{protocol_options,<<"no_sslv3|no_sslv2">>},{dhfile,<<"/etc/ssl/private/ejabberd/dh2048.pem">>},{certfile,<<"/etc/ssl/private/wildcard.example.com-with-intermediate-and-key.crt">>},{ciphers,<<"ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES256-SHA384:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA:ECDHE-RSA-AES256-SHA:AES128-GCM-SHA256:AES256-GCM-SHA384:AES128-SHA256:AES256-SHA256:AES128-SHA:AES256-SHA:!DSS">>}],true,{jid,<<"user_person">>,<<"example.com">>,<<"mox">>,<<"user_person">>,<<"example.com">>,<<"mox">>},<<"user_person">>,<<"example.com">>,<<"mox">>,{{1485,212322,88610},<0.20956.69>},{pres_t,24},{pres_f,24},{pres_a,24},{xmlel,<<"presence">>,[],[{xmlel,<<"show">>,[],[{xmlcdata,<<"away">>}]},{xmlel,<<"status">>,[],[{xmlcdata,<<"you
took down captain bob's steering
wheel?">>}]},{xmlel,<<"c">>,[{<<"xmlns">>,<<"http://jabber.org/protocol/caps">>},{<<"node">>,<<"http://pidgin.im/">>},{<<"hash">>,<<"sha-1">>},{<<"ver">>,<<"LnqHY7NoTe66rRUAqUoaHlaYL6s=">>},{<<"ext">>,<<"voice-v1">>}],[]},{xmlel,<<"x">>,[{<<"xmlns">>,<<"vcard-temp:x:update">>}],[{xmlel,<<"photo">>,[],[{xmlcdata,<<"b92f1b559265db64f70cc1281c5ea6e71fd04063">>}]}]}]},{1485,212323,463135},{userlist,none,[],false},c2s_tls,ejabberd_auth_external,{{192,241,191,224},39383},[{caps_resources,{1,{{<<"user_person">>,<<"example.com">>,<<"mox">>},{caps,<<"http://pidgin.im/">>,<<"LnqHY7NoTe66rRUAqUoaHlaYL6s=">>,<<"sha-1">>,[<<"voice-v1">>]},nil,nil}}}],active,[],inactive,undefined,undefined,1000,undefined,300,300,false,0,0,<<>>}
** Reason for termination =
**
{badarg,[{xml,element_to_binary,[{xmlel,<<"message">>,[{<<"from">>,<<"user_person@example.com">>},{<<"to">>,<<"user_person@example.com/mox">>},{<<"type">>,<<"headline">>}],[{xmlel,<<"event">>,[{<<"xmlns">>,<<"http://jabber.org/protocol/pubsub#event">>}],[{xmlel,<<"items">>,[{<<"node">>,<<"http://jabber.org/protocol/nick">>}],[{xmlel,<<"item">>,[{<<"id">>,"5704CDFE9EEEC"}],[{xmlelement,"nick",[{"xmlns","http://jabber.org/protocol/nick"}],[{xmlcdata,<<"Tom
Fiddle">>}]}]}]}]},{xmlel,<<"delay">>,[{<<"xmlns">>,<<"urn:xmpp:delay">>},{<<"from">>,<<"user_person@example.com/9951760901392301947684259">>},{<<"stamp">>,<<"2014-02-13T18:07:58Z">>}],[{xmlcdata,<<>>}]}]}],[]},{ejabberd_c2s,send_element,2,[{file,"src/ejabberd_c2s.erl"},{line,1856}]},{ejabberd_c2s,send_stanza,2,[{file,"src/ejabberd_c2s.erl"},{line,1866}]},{ejabberd_c2s,handle_info,3,[{file,"src/ejabberd_c2s.erl"},{line,1730}]},{p1_fsm,handle_msg,10,[{file,"src/p1_fsm.erl"},{line,582}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,240}]}]}
2017-01-23 22:58:43 =CRASH REPORT====
crasher:
> initial call: gen:init_it/6
> pid: <0.20956.69>
> registered_name: []                                                                                                                                                                 
> exception exit:                                                                                                                                                                     
{{badarg,[{xml,element_to_binary,[{xmlel,<<"message">>,[{<<"from">>,<<"user_person@example.com">>},{<<"to">>,<<"user_person@example.com/mox">>},{<<"type">>,<<"headline">>}],[{xmlel,<<"event">>,[{<<"xmlns">>,<<"http://jabber.org/protocol/pubsub#event">>}],[{xmlel,<<"items">>,[{<<"node">>,<<"http://jabber.org/protocol/nick">>}],[{xmlel,<<"item">>,[{<<"id">>,"5704CDFE9EEEC"}],[{xmlelement,"nick",[{"xmlns","http://jabber.org/protocol/nick"}],[{xmlcdata,<<"Tom                                                                                       
Fiddle">>}]}]}]}]},{xmlel,<<"delay">>,[{<<"xmlns">>,<<"urn:xmpp:delay">>},{<<"from">>,<<"user_person@example.com/9951760901392301947684259">>},{<<"stamp">>,<<"2014-02-13T18:07:58Z">>}],[{xmlcdata,<<>>}]}]}],[]},{ejabberd_c2s,send_element,2,[{file,"src/ejabberd_c2s.erl"},{line,1856}]},{ejabberd_c2s,send_stanza,2,[{file,"src/ejabberd_c2s.erl"},{line,1866}]},{ejabberd_c2s,handle_info,3,[{file,"src/ejabberd_c2s.erl"},{line,1730}]},{p1_fsm,handle_msg,10,[{file,"src/p1_fsm.erl"},{line,582}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,240}]}]},[{p1_fsm,terminate,8,[{file,"src/p1_fsm.erl"},{line,760}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,240}]}]}
> ancestors: [<0.1072.0>,ejabberd_listeners,ejabberd_sup,<0.37.0>]
> messages: []
> links: [#Port<0.1810780>]
> dictionary: [{'$internal_queue_len',0},{random_seed,{11954,825,4265}}]
> trap_exit: false
> status: running
> heap_size: 6772
> stack_size: 27
> reductions: 36089
neighbours:
````

**Comment by badlop** [comment]
I see a mention to xmlelement in the crash log. This was the old packet format used in ejabberd 2, that should have been updated automatically to the new xmlel format, obviously there was some problem in the database migration.

The most simple solution, if you don't bother about the pubsub information stored, is to delete the pubsub table contents. For that, if you use the default internal Mnesia database: Enter the ejabberd WebAdmin page -> Nodes -> your node -> Database -> for the pubsub tables, swtich to "delete content" -> then click "Send".

**Comment by zinid** [comment]
Fixed in 17.01

**Comment by sudoman** [comment]
For anyone who stumbles upon this issue, here is the db upgrade method in case you want to go back to using the version of ejabberd in your distro: https://github.com/processone/ejabberd/issues/1409#issuecomment-271549380
